apps/challenges/models.py

```diff
@@ -73,6 +80,14 @@ def is_active(self):
         return False
 
 
+@receiver(signals.post_save, sender=Challenge)
```

Comment:
The reason for using signal was a common function which can be applied across all models. This function is ok but I will really be more happy if you could somehow dig and see if `create_post_challenge` and `create_post_challengephase` can be made a single function.